libs/localization/src/lib/polyfills.ts

```diff
@@ -35,35 +33,33 @@ const localeDataModules = {
   },
 }
 
-const maybePolyfillNumberFormat = (locale: Locale) => {
-  // Polyfill Intl.NumberFormat if necessary
-  if (!areIntlLocalesSupported(locale) || shouldPolyfillNumberFormat()) {
-    return import(
-      /* webpackChunkName: "intl-numberformat" */ '@formatjs/intl-numberformat/polyfill-force'
-    )
-  }
+const maybePolyfillLocale = async () => {
+  await import(
+    /* webpackChunkName: "intl-locale" */ '@formatjs/intl-locale/polyfill'
+  )
 }
 
-const maybePolyfillDateTimeFormat = (locale: Locale) => {
-  // Polyfill Intl.DateTimeFormat if necessary
-  if (!areIntlLocalesSupported(locale)) {
-    return import(
-      /* webpackChunkName: "intl-datetimeformat" */ '@formatjs/intl-datetimeformat/polyfill-force'
-    )
-  }
+const maybePolyfillNumberFormat = async () => {
+  await import(
+    /* webpackChunkName: "intl-numberformat" */ '@formatjs/intl-numberformat/polyfill'
+  )
+}
+
+const maybePolyfillDateTimeFormat = async () => {
+  await import(
+    /* webpackChunkName: "intl-datetimeformat" */ '@formatjs/intl-datetimeformat/polyfill'
+  )
```

Comment:
A few issues:

1. Major: It is common for browsers to support Intl and uncommon for them to support Icelandic. For us to polyfill Icelandic, we need to force-load the polyfill first (with polyfill-force). I.e. this may break the Icelandic locale on Chrome.
2. Minor: We would like to shave of some KBs if the browser supports Intl AND the chosen locale.
3. Minor: If we're always going to load all the polyfills, we should not dynamically import them from multiple different chunks. Either just import them normally, or configure the same chunk name for all of them.